src/moai-android-adcolony/host.cpp

```diff
@@ -1,6 +1,16 @@
 // Copyright (c) 2010-2011 Zipline Games, Inc. All Rights Reserved.
 // http://getmoai.com
 
+// From moai-box2d's pch.h
+#include <moai_config.h>
+#include <zl-util/headers.h>
+#include <moai-core/headers.h>
+#include <moai-sim/headers.h>
```

Comment:
The way the android build works it to always compile all host modules to their static libs, then in the host gradle file you pick which android plugins you want and it will link them in to the binary. There should be no massive IFDEF around the code, that is a legacy thing/regression. The new android host module system is as easy as adding "moai-android-adcolony" to a config line and pressing rebuild in android studio to add moai-android-adcolony support